buildcache/build_cache_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package buildcache
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/arduino/go-paths-helper"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_UpdateLastUsedFileNotExisting(t *testing.T) {
+
+	testBuildDir := paths.New(t.TempDir(), "sketches", "sketch-xxx")
+	err := os.MkdirAll(testBuildDir.String(), 0770)
+	require.Nil(t, err)
+	timeBeforeUpdating := time.Unix(0, 0)
+	requireCorrectUpdate(t, testBuildDir, timeBeforeUpdating)
+}
+
+func Test_UpdateLastUsedFileExisting(t *testing.T) {
+
+	testBuildDir := paths.New(t.TempDir(), "sketches", "sketch-xxx")
+	err := os.MkdirAll(testBuildDir.String(), 0770)
+	require.Nil(t, err)
+
+	// create the file
+	preExistingFile := testBuildDir.Join(lastUsedFileName).String()
+	err = paths.New(preExistingFile).WriteFile([]byte{})
+	require.Nil(t, err)
+	timeBeforeUpdating := time.Now().Add(-time.Second)
+	os.Chtimes(preExistingFile, time.Now(), timeBeforeUpdating)
+	requireCorrectUpdate(t, testBuildDir, timeBeforeUpdating)
+}
+
+func requireCorrectUpdate(t *testing.T, dir *paths.Path, prevModTime time.Time) {
+	err := Used(dir)
+	require.Nil(t, err)
+	expectedFile := dir.Join(lastUsedFileName)
+	fileInfo, err := os.Stat(expectedFile.String())
+	require.Nil(t, err)
+
+	require.GreaterOrEqual(t, fileInfo.ModTime(), prevModTime)
+}
+
+func TestPurge(t *testing.T) {
+	ttl := time.Minute
+
+	dirToPurge := paths.New(t.TempDir(), "root")
+
+	lastUsedTimesByDirPath := map[*paths.Path]time.Time{
+		(dirToPurge.Join("old")):   time.Now().Add(-ttl - time.Hour),
+		(dirToPurge.Join("fresh")): time.Now().Add(-ttl + time.Minute),
+	}
+
+	// create the metadata files
+	for dirPath, lastUsedTime := range lastUsedTimesByDirPath {
+		err := os.MkdirAll(dirPath.Canonical().String(), 0770)
+		require.Nil(t, err)
+		infoFilePath := dirPath.Join(lastUsedFileName).Canonical().String()
+		err = paths.New(infoFilePath).WriteFile([]byte{})
+		require.Nil(t, err)
+		// make sure access time does not matter
+		accesstime := time.Now()
+		err = os.Chtimes(infoFilePath, accesstime, lastUsedTime)
+		require.Nil(t, err)
+	}
+
+	Purge(dirToPurge, ttl)
+
+	fileinfo, err := os.Stat(dirToPurge.Join("fresh").String())
+	require.Nil(t, err)
+	require.True(t, fileinfo.IsDir())
+	_, err = os.Stat(dirToPurge.Join("old").String())
+	require.ErrorIs(t, err, os.ErrNotExist)
```

Comment:
Change all calls to `os` with their `go-paths` equivalent:
`os.MkdirAll()` -> `pathObj.MkdirAll()`
`os.Chtimes()` -> `pathObj.Chtimes()`
`os.Stat()` -> `pathObj.Stat()`
`os.Stat()` + `err != os.ErrNotExist` -> `pathObj.Exist()`